packages/binding-coap/src/coap-server.ts

```diff
@@ -182,17 +185,64 @@ export default class CoapServer implements ProtocolServer {
         return `${this.scheme}://${address}:${port}/${encodeURIComponent(urlPath)}`;
     }
 
+    private fillInMetaPropertiesBindingData(thing: ExposedThing, base: string, offeredMediaType: string) {
+        const opValues = this.createPropertyMetaOpValues(thing);
+
+        if (opValues.length === 0) {
+            return;
+        }
+
+        if (thing.forms == null) {
+            thing.forms = [];
+        }
+
+        const form = this.createAffordanceForm(base, this.PROPERTY_DIR, offeredMediaType, opValues, thing.uriVariables);
+
+        thing.forms.push(form);
+    }
+
+    private getReadableProperties(thing: ExposedThing) {
+        return Object.entries(thing.properties).filter(([_, value]) => value.writeOnly !== true);
+    }
+
+    private getWritableProperties(thing: ExposedThing) {
+        return Object.entries(thing.properties).filter(([_, value]) => value.readOnly !== true);
+    }
+
+    private createPropertyMetaOpValues(thing: ExposedThing): string[] {
+        const properties = Object.values(thing.properties);
+        const numberOfProperties = properties.length;
+
+        if (numberOfProperties === 0) {
+            return [];
+        }
+
+        const readableProperties = this.getReadableProperties(thing).length;
+
+        const opValues: string[] = [];
+
+        if (readableProperties > 0) {
+            opValues.push("readmultipleproperties");
+        }
+
+        if (readableProperties === numberOfProperties) {
+            opValues.push("readallproperties");
+        }
```

Comment:
Actually, the spec is not clear about this semantic. One could still understand that the readallproperties operation just works on the set of readable properties of the thing. I guess for the time being is alright to leave as it is, but still other bindings implementations might act differently.